tests/model_servering/server/llm/conftest.py

```diff
@@ -0,0 +1,177 @@
+import pytest
+from kubernetes.dynamic import DynamicClient
+from kubernetes.dynamic.exceptions import NotFoundError
+from ocp_resources.inference_service import InferenceService
+from ocp_resources.namespace import Namespace
+from ocp_resources.resource import ResourceEditor, get_client
+from ocp_resources.secret import Secret
+from ocp_resources.service_account import ServiceAccount
+from ocp_resources.service_mesh_member_roll import ServiceMeshMemberRoll
+from ocp_resources.serving_runtime import ServingRuntime
+from pytest_testconfig import config as py_config
+
+from tests.model_servering.server.llm.utils import base64_encode_str
+from utilities.constants import TRANSFORMERS_CACHE_ENV_VAR
+from utilities.inference_service import InferenceServiceForTests
+
+
+@pytest.fixture(scope="session")
+def admin_client() -> DynamicClient:
+    return get_client()
+
+
+@pytest.fixture(scope="class")
+def model_namespace(request, admin_client: DynamicClient) -> Namespace:
+    with Namespace(
+        client=admin_client,
+        name=request.param["name"],
+    ) as ns:
+        ns.wait_for_status(status=Namespace.Status.ACTIVE, timeout=120)
+        yield ns
+
+
+@pytest.fixture(scope="class")
+def patched_service_mesh_member_roll(admin_client: DynamicClient, model_namespace: Namespace) -> ServiceMeshMemberRoll:
+    smmr = ServiceMeshMemberRoll(client=admin_client, namespace="istio-system", name="default")
+
+    if not smmr.exists:
+        raise NotFoundError("istio ServiceMeshMemberRoll does not exist")
+
+    members = smmr.instance.spec.members
+
+    if members:
+        members.append(model_namespace.name)
+    else:
+        members = [model_namespace.name]
+
+    with ResourceEditor(patches={smmr: {"spec": {"members": members}}}):
+        yield smmr
+
+
+@pytest.fixture(scope="class")
+def storage_uri(request) -> str:
+    return f"s3://{py_config['model_s3_bucket_name']}/{request.param['model-dir']}/"
+
+
+@pytest.fixture(scope="class")
+def llm_custom_isvc(
+    request, admin_client: DynamicClient, model_namespace: Namespace, storage_uri: str
+) -> InferenceServiceForTests:
+    name = model_namespace.name
+    with InferenceServiceForTests(
+        client=admin_client,
+        namespace=name,
+        name=name,
+        deployment_mode=request.param["deployment-mode"],
+        predictor_service_account_name="models-bucket-sa",
+        predictor_model_format_name="pytorch",
+        predictor_storage_uri=storage_uri,
+    ) as isvc:
+        yield isvc
+
+
+@pytest.fixture(scope="class")
+def endpoint_s3_secret(
+    admin_client: DynamicClient,
+    model_namespace: Namespace,
+    aws_access_key: str,
+    aws_secret_access_key: str,
+) -> Secret:
+    data = {
```

Comment:
if i understood correctly we are pulling this from env , imo it would be good we we and those variable in yaml file 
because that is more readable and more easy to execute

Reply:
It's actually defined as `access_key = py_config.get("aws_access_key", os.environ.get("AWS_ACCESS_KEY_ID"))`, so it tries to get it from the py_config singleton object or defaults to the env variable, raising an exception if neither is found. 
This seems more elegant than a YAML file to me

Reply:
you can overwrite the value by passing `tc=aws_access_key:xxxx`

you set up an env for tests, so it should be part of the setup. we can extend to using also `.aws/credentials` etc but not add additional config

Reply:
and as @lugi0 mentioned - the values can be set in global_config.py

Reply:
Here are my two cents
The values in global_config.py will be hard-coded, 
I would suggest using JSON, YAML, or any other data serialization format. My goal is to make the configuration dynamic and user-friendly.

Reply:
where would the config in the file be populated from?